### DISCUSSION TITLE
How to define a schema for an object with multiple key value pairs?

### DISCUSSION THREAD

**Discussion by pawan-kr1997** [prisma/prisma#17605] [Q&A] [answered]
I am using MongoDB as my database. I have a variable called ``bookmark``. It is an object that contain multiple key-value pairs. ``key`` is the id of the bookmarked post and ``value`` is the data of the post.

Here is an example:
```
bookmark={
      "1":{
                id:"1",
                title:"hello"
            },
      "2":{
               id:"2",
               title:"world"
           },
}
```
The bookmark data is not static i.e new key-value pair could be added or removed. Please guide me

**Comment by nurul3101** [comment, marked as answer]
Hello @pawan-kr1997 👋 

In this case, it looks like the keys are dynamic, so I believe you would need to use JSON to store dynamic keys.
Your model could look something like this:

```prisma
model Books {
  id        String @id @default(auto()) @map("_id") @db.ObjectId
  bookmarks Json?
}
```
Here's a reference for using [JSON Fields](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
